connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/IncrementalCooperativeAssignor.java

```diff
@@ -285,18 +292,13 @@ ClusterAssignment performTaskAssignment(
 
         handleLostAssignments(lostAssignments, newSubmissions, completeWorkerAssignment);
 
+        if (delay > 0) {
+            log.debug("Delaying {}ms for revoking tasks.", delay);
+        }
         // Do not revoke resources for re-assignment while a delayed rebalance is active
-        // Also we do not revoke in two consecutive rebalances by the same leader
-        canRevoke = delay == 0 && canRevoke;
-
-        // Compute the connectors-and-tasks to be revoked for load balancing without taking into
-        // account the deleted ones.
-        log.debug("Can leader revoke tasks in this assignment? {} (delay: {})", canRevoke, delay);
-        if (canRevoke) {
+        if (delay == 0) {
             Map<String, ConnectorsAndTasks> toExplicitlyRevoke =
-                    performTaskRevocation(activeAssignments, currentWorkerAssignment);
-
-            log.debug("Connector and task to revoke assignments: {}", toRevoke);
+                    performTaskRevocation(configured, completeWorkerAssignment);
```

Comment:
Yes that's exactly the point. I got this logic from the original PR from @showuon  here => https://github.com/apache/kafka/pull/10367 . To quote =>

```
we also passed the configured assignment into performTaskRevocation instead of activeAssignment, so that we can compute the correct expected max assignment number (totalSize / workerSize), instead of the activeTotalSize / workerSize, because the activeTotalSize doesn't include the newAssignments, which will cause the wrong computation, and cause uneven rebalance, or need more round of revoking rebalance.
```

I also noticed that if I don't pass configured, then post allowed + successive revoking rebalance, the assignments were  not balanced.

Reply:
``` 
But I'm just wondering if there's a reason why we were doing revocations only based on existing member assignments earlier.
```
Regarding this I dont know tbh, but my guess is since earlier we were not doing successive revocations, it was not needed. @C0urante can certainly throw some more light on this.

Reply:
@yashmayya , yes, your analysis is correct, I was trying to take the new added/deleted connectors/tasks into account for the revocation calculation, because I was trying to do consecutive revocation without delay.

> But I'm just wondering if there's a reason why we were doing revocations only based on existing member assignments earlier.

I guess that's because we didn't consider too complicated scenarios. Anyway, I think this change makes sense.

Reply:
Thanks @vamossagar12 and @showuon, I do agree that this change makes sense!

Reply:
Switching from `configured` to `activeAssignments` helps in some scenarios, but not others. I found a few cases to demonstrate this while reviewing https://github.com/apache/kafka/pull/10367, documented [here](https://github.com/apache/kafka/pull/10367#discussion_r810393411).

I wonder if we can leave this change out for now and then address the remaining issues with incremental rebalancing in a follow-up PR? There are a lot of edge cases to consider and I don't want to block the primary change here of allowing consecutive revocations (which is super useful on its own!) by tying it to thinking through all the implications for a change that can be made independently.

Reply:
@C0urante  When you say `I wonder if we can leave this change out for now` do you mean retaining this or switching back to using `activeAssignments` instead of `configured`? I will also take a look at the scenario you described about where this might fail. If we want to retain this, I can take a look at them in a separate PR or this one, whatever is decided over here or if we want to switch back to `activeAssignments`, then that would need more work probably in this PR.

Reply:
I mean switching back to using `activeAssignments`.

Reply:
@showuon let me know what you think about this

Reply:
I'm good to revert back to `activeAssignments` and discuss in other PR or JIRA. After all, it's not relevant to this issue. Thanks

Reply:
Thanks @showuon I reverted it back.